piptools/utils.py

```diff
@@ -45,10 +27,10 @@ def key_from_ireq(ireq):
 def key_from_req(req):
     """Get an all-lowercase version of the requirement's name."""
     if hasattr(req, 'key'):
-        # pip 8.1.1 or below, using pkg_resources
+        # from pkg_resources, such as installed dists for pip-sync
```

Comment:
I believe this will be stable now that we use vendored pip

Reply:
It thought so too, but unfortunately there's another condition unrelated to the pip version.
When parsing the *installed* packages (in pip-sync), the returned object by `pip.get_installed_distributions` is a `DistributionEggInfo` or something alike, which comes from `pkg_resources`, and those object do not have a `.name` attribute.
The `if hasattr(req, 'key')` condition masked that case, making us believe it was only a pip8-9 difference.

Reply:
ok